### ISSUE TITLE
Crash in Molten Core if you kill Garr before any of his adds

### ISSUE THREAD

**Issue by TQPS** [trickerer/Trinity-Bots#209]
If you kill Garr before you take down any adds, this crash occurs.

If you kill Garr AFTER most of the adds are down, this crash does NOT occur.

[Crash Log](https://gist.github.com/TQPS/3a1cc21bd2b79e97cbbf563172a31848)

For what it's worth:
- I am level capped at 60 (via Stormwind NPC)
- I use 39 bots
- AtlasLoot is the only add on I use
- No other custom changes or alterations to server other than NPCBots
- Server is Windows 7 / Client is Windows 10

Replicate:
-----------
1. Go into Molten core and clear up to Garr
2. Tell bots to target Garr, ignore adds (Warlocks will banish)
3. When Garr is almost at 0 health or dead, the game should freeze in place whilst animations continue and server will pop a crash message.
4. Log back into game - you're still in Molten Core, Garr's body is not there and usually 1+ add remains on his spawn

Workaround:
--------------
1. Go into Molten core and clear up to Garr
2. Tell bots to target adds, ignore Garr
3. When adds are down, focus Garr
4. Fight ends, Garr is dead, no crash, Garr can be looted.

[Issue logged with TC](https://github.com/TrinityCore/TrinityCore/issues/28274)
Was advised issue is with the bots:

**Comment by trickerer** [comment]
Well the TC comment sent you here rightfully, the crash **is** caued by bots.

I noticed in crashlog local variables are related to autoloot.
Couldn't reproduce it after 5 tries, despite setting all bots to loot everything.

You seem not to be using latest bots, so I need to know what your `bot.cpp line 15063` looks like

**Comment by TQPS** [comment]
Thanks for looking into this.

I update each built from your merged 3.3.5+bots repo and then pull (if) any differences from TC 3.3.5.

I could not find a bot.cpp, did you mean bot_ai,cpp? If so line 15063 is as follows:

![image](https://user-images.githubusercontent.com/66294837/191857673-2d521b2a-ef36-4a29-ac0d-4863bcf600e5.png)

**Comment by TQPS** [comment]
For additional context, my bots dont/shouldn't loot. I remember turning that on ages ago and it was annoying as hell. Only I loot.

That aside, Garr is the only boss in MC this happens on.

**Comment by trickerer** [comment]
So, I took 20 druids with me and it didn't change anything.

You see, the only way it could crash really is related to those Firesworn instantly despawning at death. But unit's own spell targets are rechecked at every update so `target` is supposed to be guaranteed to be always valid if map updates run using single thread (which is default mode).
Please check your config and see if `MapUpdate.Threads` is not equal to `1`.

**Comment by TQPS** [comment]
I can see in my worldserver.conf that MapUpdate.Threads is equal to 1.

```
#    MapUpdate.Threads
#        Description: Number of threads to update maps.
#        Default:     1

MapUpdate.Threads = 1
```

If you cant replicate it don't worry about it. Doesn't look like anyone else is having the problem. The work around is to kill the adds first.

**Comment by TQPS** [comment]
TrinityCore rev. 677609692f09 2022-10-14 21:45:41 +0000 (npcbots_3.3.5 branch) (Win64, RelWithDebInfo, Static) 

Update: reran MC tonight with 39 bots and this time, I told the Warlocks (I use 5 of them) to NOT use Banish.

Targeted Garr first - he went down, took adds down, no crash. Got a capture of the fight for future reference.

Instead of using banish, the bots used fear to control the adds.

I will re-enable banish after server resets and do another run.

**Comment by TQPS** [comment]
TrinityCore rev. 677609692f09 2022-10-14 21:45:41 +0000 (npcbots_3.3.5 branch) (Win64, RelWithDebInfo, Static)

Update: reran MC today with my 39 bots and I allowed Warlocks to use banish.

Targeted Garr first - he went down, and just as we were switching to his adds, server crashed like clockwork.

[Crash log](https://gist.github.com/TQPS/7661b655a644809516ae395943f038ab)

TL;DR:
Warlock Bots using Banish = Server crashes after Garr down
Warlock Bots NOT using Banish = Server is fine after Garr down

Unfortunately, my videos are too large to upload here.
